src/main/java/org/ieeervce/gatekeeper/Entity/User.java

```diff
@@ -0,0 +1,44 @@
+package org.ieeervce.gatekeeper.Entity;
+
+
+import jakarta.persistence.*;
+
+import java.time.LocalDateTime;
+
+@Entity
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long userId;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(unique = true,nullable = false)
+    private String email;
+
+    @Column(nullable = false)
+    private String password;
+
+
+    @ManyToOne
+    @JoinColumn(name = "role", referencedColumnName = "roleName")
+    private Role role;
+
+    @ManyToOne
+    @JoinColumn(name = "society", referencedColumnName = "name",nullable = true)
```

Comment:
This would not be a valid. This implies that the society id points to `Society.name` to uniquely identify the society.

I would recommend using the designated id (currently `userId`, suggested to rename to `societyId`)


---

As of now, running in Springboot, I see this error:
`org.hibernate.tool.schema.spi.CommandAcceptanceException: Error executing DDL "alter table user add constraint FKhyhqoiq187lnhceqgu4vxyx79 foreign key (society) references society (name)" via JDBC [Failed to add the foreign key constraint. Missing index for constraint 'FKhyhqoiq187lnhceqgu4vxyx79' in the referenced table 'society']`